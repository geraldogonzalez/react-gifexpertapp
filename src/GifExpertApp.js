import React, { useState } from 'react';

import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

const GifExpertApp = ({ defaultCategories = []}) => {

  // const [categories, setCategories] = useState(['Dragon Ball']);
  const [categories, setCategories] = useState(defaultCategories);

  // const handleAdd = () => {
    
  //   /* const newCategories = [...categories];
  //   setCategories([...newCategories, 'Naruto']); */

  //   // setCategories(['Naruto', ...categories]);

  //   setCategories( cats => [...cats, 'Naruto']);
  // }

  return (
    <>
      <h2>GifExpertApp</h2>
      <hr />

      <AddCategory setCategories={ setCategories } />

      <ol>
        {

          categories.map(category => (

              <GifGrid 
                key={ category }
                category={ category } 
              />

            )
          )

        }
      </ol>
    </>
  )

}

export default GifExpertApp;
