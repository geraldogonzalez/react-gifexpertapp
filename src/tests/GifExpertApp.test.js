import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas sobre el componente <GifExpertApp />', () => {
        
    test('Debe mostrarse correctamente', () => {

        const wrapper = shallow( <GifExpertApp /> );

        expect( wrapper ).toMatchSnapshot();

    })
    
    test('Debe mostrar una lista de categorÃ­as', () => {
        
        const categories = ['One Punch', 'Dragon Ball Z'];
        const wrapper = shallow( <GifExpertApp defaultCategories={ categories } /> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );

    })
    

})
