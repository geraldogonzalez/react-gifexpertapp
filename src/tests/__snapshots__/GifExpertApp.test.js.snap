// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente <GifExpertApp /> Debe mostrar una lista de categor√≠as 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball Z"
      key="Dragon Ball Z"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas sobre el componente <GifExpertApp /> Debe mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;
